students/ColeP/Session09/mailroom_oo.py

```diff
@@ -0,0 +1,185 @@
+#  in class switch dictionary stuff
+
+# def one():
+#     return 'one'
+#
+#
+# def main():
+#
+#     switch = {
+#         '1': one,
+#         '2': ...,
+#         '3': ...,
+#         '4': ...
+#     }
+#
+#     while True:
+#         choice = input('1-4')
+#         if choice not in switch.keys():
+#             print('select 1-4')
+#         else:
+#             switch[choice]()
+donors_old = {"Donny Donor": [100, 10, 45],
+          "Gav Giver": [12, 16, 20],
+          "Stingy Steve": [2, 5, 1],
+          "Freaky Frank": [1200, 999, 1005],
+          }
+
+
+class Donor:
+    def __init__(self, name, donations=None):
+        self.norm_name = self.normalize_name(name)
+        self.name = name.strip()
+
+        if donations is None:
+            self.donations = []
+        else:
+            try:
+                self.donations = list(donations)
+            except:  # wants a specific error message?
+                self.donations = [donations]
+
+        # self.ty = '''
+        # Dear {},
+        #
+        # Thank you for your most recent donation of ${:.2f} to The Org!
+        #
+        # -The Org'''.format(name, donations)
+
+    def add_donation(self, donation):
+        self.donations += [donation]
+
+    def dictionize(self):
+        return {self.name: self.donations}
+
+    def ty(self):
+        return ('''
+        Dear {},
+
+        Thank you for your most recent donation of ${:.2f} to The Org!
+
+        -The Org'''.format(self.name, self.donations[-1]))
+
+    @staticmethod
+    def normalize_name(name):
+        return name.lower().strip().replace(' ', '')
+
+
+class StatsTable:
+    def __init__(self, in_dict):
+        self.in_dict = in_dict
+
+    @staticmethod
+    def header():
+        return "{:22s} _ {:11s} _ {:9s} _ {:12s}".format("Donor Name", "Total Given", "Num Gifts", "Average Gift")
+
+    @staticmethod
+    def take_total(item):  # used in conjunction with sorted() to take the donations and sort by total donation
+        return item[1]
+
+    def body(self):
+        compy = [(donor, sum(self.in_dict[donor]), len(self.in_dict[donor]),
+                  sum(self.in_dict[donor]) / len(self.in_dict[donor])) for donor in self.in_dict]
+        sorted_donors = sorted(compy, key=self.take_total, reverse=True)
+        # sorts the donor list by their contribution totals
+
+        empt = ''
+        for row in sorted_donors:
+            empt += ('{:22s} ${:11.2f} {:^13}$ {:>12.2f}\n'.format(*row))
+        return empt
+
+    def thx4all(self):
+        for i in self.in_dict:
+            with open('{}.txt'.format(i), 'w+') as let:
+                let.writelines('''Dear {},
+
+Thank you for your total donations of ${:.2f} to The Org!
+
+-The Org'''.format(i, sum(self.in_dict[i])))
+
+# class MakeDict:
+#     def big_dict(*args):
+#         d = {}
+#         for i in args:
+#             d = {**i}
+#         return d
+
+
+def main_menu():
```

Comment:
Don't forget to actually implement your OO changes in this version of mailroom! All your methods should be using the donor models instead of this default donors dict.